src/prpy/planning/vectorfield.py

```diff
@@ -197,32 +217,165 @@ def TerminateMove():
 
             return Status.CONTINUE
 
+        integration_timelimit = 10.0
         return self.FollowVectorField(robot, vf_straightline, TerminateMove,
+                                      integration_timelimit,
+                                      timelimit, 
+                                      **kw_args)
+
+
+    @PlanningMethod
+    def PlanWorkspacePath(self, robot, traj,
+                          timelimit=5.0,
+                          position_tolerance=0.01,
+                          angular_tolerance=0.15,
+                          goal_pose_tol=0.01,
+                          **kw_args):
+        """
+        Plan a configuration space path given a workspace path.
+        Trajectory timing information is ignored.
+
+        @param openravepy.Robot      robot: The robot.
+        @param openravepy.Trajectory traj:  Workspace trajectory,
+                                            represented as an
+                                            OpenRAVE AffineTrajectory.
+        @param float timelimit: Max planning time (seconds).
+        @param float position_tolerance: Constraint tolerance (meters).
+        @param float angular_tolerance:  Constraint tolerance (radians).
+        @param float goal_pose_tol: Goal tolerance (meters).
+
+        @return openravepy.Trajectory qtraj: Configuration space path.
+        """
+        if position_tolerance < 0.0:
+            raise ValueError('Position tolerance must be non-negative.')
+        elif angular_tolerance < 0.0:
+            raise ValueError('Angular tolerance must be non-negative.')
+
+        manip = robot.GetActiveManipulator()
+        Tstart = manip.GetEndEffectorTransform()
+
+        # Get the final end-effector pose
+        duration = traj.GetDuration()
+        T_ee_goal = openravepy.matrixFromPose(traj.Sample(duration)[0:7])
+
+        def vf_path():
+            """
+            Function defining a joint-space vector field.
+            """
+            T_ee_actual = manip.GetEndEffectorTransform()
+
+            # Find where we are on the goal trajectory by finding
+            # the the closest point
+            (_,t) = util.GetMinDistanceBetweenTransformAndWorkspaceTraj(
+                                                                   T_ee_actual,
+                                                                   traj)
+            # Get the desired end-effector transform from
+            # the goal trajectory
+            t_step = 0.2
+            desired_T_ee = openravepy.matrixFromPose(traj.Sample(t)[0:7])
+            pose_ee_next = traj.Sample(t+t_step)[0:7]
```

Comment:
It looks like you are using `t_step` to approximate a finite-difference velocity. This is problematic because:

1. the trajectory is un-timed (see above)
2. you have no guarantee on the scale of the parameterization - 0.2 may be very large or very small

Ideally, we would directly extract the derivative from the trajectory and use this as the feed-forward term in the controller. If that is not possible, then we should at least convert the trajectory to a known parametrization (e.g. unit velocity) so we know the scale of `t_step`.